cmd/dowload_test.go

```diff
@@ -124,6 +130,57 @@ func TestRunDownloadFailsIntegrityTest(t *testing.T) {
 	assert.Contains(t, err.Error(), "integrity mismatch")
 }
 
+main
+func TestOptimization(t *testing.T) {
+	// Setup test server
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("test content"))
+	}))
+	defer ts.Close()
+
+	t.Run("Valid File Not Redownloaded", func(t *testing.T) {
+		tmpDir := test.TmpDir(t)
+		lockDir := filepath.Join(tmpDir, "valid_test")
+		err := os.MkdirAll(lockDir, 0755)
+		require.NoError(t, err)
+
+		testUrl := ts.URL + "/valid_test.txt"
+		testFile := test.TmpFile(t, "test content")
+
+		lockPath := test.TmpFile(t, "")
+		lock, err := internal.NewLock(lockPath, true)
+		require.NoError(t, err)
+
+		err = lock.AddResource([]string{testUrl}, internal.RecommendedAlgo, nil, filepath.Base(testFile))
+		require.NoError(t, err)
+
+		err = lock.Download(tmpDir, nil, nil, "")
+		require.NoError(t, err)
+	})
+
+	t.Run("Invalid File Redownloaded", func(t *testing.T) {
+		tmpDir := test.TmpDir(t)
+		lockDir := filepath.Join(tmpDir, "invalid_test")
+		err := os.MkdirAll(lockDir, 0755)
+		require.NoError(t, err)
+
+		testUrl := ts.URL + "/invalid_test.txt"
+		testFile := test.TmpFile(t, "test content")
+
+		lockPath := test.TmpFile(t, "")
+		lock, err := internal.NewLock(lockPath, true)
+		require.NoError(t, err)
+
+		err = lock.AddResource([]string{testUrl}, internal.RecommendedAlgo, nil, filepath.Base(testFile))
+		require.NoError(t, err)
+
+		err = os.WriteFile(testFile, []byte("corrupted"), 0644)
+		require.NoError(t, err)
+
+		err = lock.Download(tmpDir, nil, nil, "")
+		require.NoError(t, err)
```

Comment:
Didn't we agree that in this case the code should error? That grabit should refuse to overwrite an incorrect file...?